src/cellars/cellar_v2_5.rs

```diff
@@ -222,6 +224,127 @@ pub fn get_encoded_function(call: FunctionCall, cellar_id: String) -> Result<Vec
 
             Ok(CellarWithMultiAssetDepositV1Calls::DropAlternativeAssetData(call).encode())
         }
+        Function::SetSharePriceOracle(params) => {
+            validate_oracle(&cellar_id, &params.registry_id, &params.share_price_oracle)?;
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetSharePriceOracleCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetSharePriceOracleCall {
+                registry_id: string_to_u256(params.registry_id)?,
+                share_price_oracle: sp_call_parse_address(params.share_price_oracle)?,
+            };
+
+            Ok(CellarV2_5Calls::SetSharePriceOracle(call).encode())
+        }
+        Function::CachePriceRouter(params) => {
+            validate_cache_price_router(
+                &cellar_id,
+                params.check_total_assets,
+                params.allowable_range,
+                Some(params.expected_price_router.clone()),
+            )?;
+            log_cellar_call(
+                CELLAR_NAME,
+                &CachePriceRouterCall::function_name(),
+                &cellar_id,
+            );
+            let call = CachePriceRouterCall {
+                check_total_assets: params.check_total_assets,
+                allowable_range: params.allowable_range as u16,
+                expected_price_router: sp_call_parse_address(params.expected_price_router)?,
+            };
+
+            Ok(CellarV2_5Calls::CachePriceRouter(call).encode())
+        }
+        Function::AddAdaptorToCatalogue(params) => {
+            validate_new_adaptor(&cellar_id, &params.adaptor, &V2_5_PERMISSIONS)?;
+            log_cellar_call(
+                CELLAR_NAME,
+                &AddAdaptorToCatalogueCall::function_name(),
+                &cellar_id,
+            );
+
+            if let Err(err) = check_blocked_adaptor(&params.adaptor) {
```

Comment:
I think this `if` check made it in here by accident. `validate_new_adaptor` checks blocked adaptors and this references governance functions.